backend/lib/youtube-audio-url-finder/audio-file-url-finder.ts

```diff
@@ -0,0 +1,88 @@
+import { Errors } from './errors.ts';
+import ErrorGenerator from './error-generator.ts';
+import { AudioFileAdaptiveFormat, AudioFileData } from './invidious.interfaces.ts';
+
+export default class AudioFileUrlFinder {
+  private invidiousVideoUrl: string;
+
+  constructor() {
+    this.invidiousVideoUrl = '';
+  }
+
+  /**
+   * Sets the invidious video url used to receive data about a video
+   *
+   * @param invidiousVideoUrl - The invidious video url
+   */
+  setInvidiousVideoUrl(invidiousVideoUrl: string): void {
+    this.invidiousVideoUrl = invidiousVideoUrl;
+  }
+
+  /**
+   * Finds an audio file url (googlevideo) for the currently set invidious video url
+   *
+   * @returns The audio file url
+
+   *
+   * @throws Errors.UNEXPECTED_OR_NO_RESPONSE
+   * Thrown if no response or an invalid response was received
+   *
+   * @throws Errors.NO_SUITABLE_ADAPTIVE_FORMATS
+   * Thrown if no supported audio format was received
+
+   */
```

Comment:
```suggestion
   * Finds an audio file url (googlevideo) for the currently set invidious video url
   *
   * @returns The audio file url
   *
   * @throws Errors.UNEXPECTED_OR_NO_RESPONSE
   * Thrown if no response or an invalid response was received
   *
   * @throws Errors.NO_SUITABLE_ADAPTIVE_FORMATS
   * Thrown if no supported audio format was received
   */
```
I hate this WebGUI....